### DISCUSSION TITLE
Auto-merge incorrectly merges PR with outstanding jobs

### DISCUSSION THREAD

**Discussion by joshlf** [community/community#69219] [Pull Requests]
### Select Topic Area

Bug

### Body

I maintain a repo with [this CI configuration](https://github.com/google/zerocopy/blob/cdead15cba4fccd5a065aa2c8c1ae6cbb933aaee/.github/workflows/ci.yml). Note that most jobs in that configuration depend on the `generate_cache` job. I have observed that, if I enable auto-merge on a PR before the `generate_cache` job has finished, the PR will be merged as soon as `generate_cache` has finished and before the remaining (now-unblocked) jobs are queued. *

My expectation is that all jobs which are configured to run on the `pull_request` trigger must complete before a PR can be merged, so I interpret this behavior as a bug.

\* Our repo's configuration is currently in flux, especially with regards to enabling/configuring the merge queue, so the repo configuration as it stands when you're reading this may not exhibit this behavior.

**Comment by Eikix** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> I maintain a repo with [this CI configuration](https://github.com/google/zerocopy/blob/cdead15cba4fccd5a065aa2c8c1ae6cbb933aaee/.github/workflows/ci.yml). Note that most jobs in that configuration depend on the `generate_cache` job. I have observed that, if I enable auto-merge on a PR before the `generate_cache` job has finished, the PR will be merged as soon as `generate_cache` has finished and before the remaining (now-unblocked) jobs are queued. *
> 
> 
> 
> My expectation is that all jobs which are configured to run on the `pull_request` trigger must complete before a PR can be merged, so I interpret this behavior as a bug.
> 
> 
> 
> \* Our repo's configuration is currently in flux, especially with regards to enabling/configuring the merge queue, so the repo configuration as it stands when you're reading this may not exhibit this behavior.

I confirm to have a very similar bug,
Except I use merge queues,

I’ve had a weird experience yesterday:
- review a PR
- approve it
- trigger auto-merge -> behaviour is: add to merge queue if all checks are a-okay
- PR gets added to merge queue 
- merge queue merges PR

problem?
the CI checks were unsuccessful -> one check was failing,
still got merged though

context:
- branch protection to the fullest 


and the job that failed was "behind" a successful job, so sequentially started after some other check finishes, just like the creator of the post
